### ISSUE TITLE
identity_vars should not pass empty email address

### ISSUE THREAD

**Issue by lihan** [jkeyes/python-intercom#100]
In ``intercom/api_operations/save.py``

The ``identity_hash`` should not allow ``email`` property with empty string if ``user_id`` presents, intercom  API returns ```BadRequestError(u"Email can't be blank",)```

**Comment by lihan** [comment]
Appended with a test case for you to reproduce
```
In [9]: user =intercom.User.create(user_id=99999)

In [10]: user.increment('logins')

In [11]: user.save()
---------------------------------------------------------------------------
BadRequestError                           Traceback (most recent call last)
<ipython-input-11-79b22bfa6db2> in <module>()
----> 1 user.save()

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/api_operations/save.pyc in save(self)
     47             # create
     48             params.update(self.identity_hash)
---> 49             response = Intercom.post('/%s' % (collection), **params)
     50         if response:
     51             return self.from_response(response)

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/__init__.pyc in post(cls, path, **params)
    169     @classmethod
    170     def post(cls, path, **params):
--> 171         return cls.request('POST', path, params)
    172 
    173     @classmethod

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/__init__.pyc in request(cls, method, path, params)
    161     def request(cls, method, path, params):
    162         return Request.send_request_to_path(
--> 163             method, cls.get_url(path), cls._auth, params)
    164 
    165     @classmethod

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/request.pyc in send_request_to_path(cls, method, url, auth, params)
     59         if resp.content and resp.content.strip():
     60             # parse non empty bodies
---> 61             return cls.parse_body(resp)
     62 
     63     @classmethod

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/request.pyc in parse_body(cls, resp)
     70             body = json.loads(decoded_body)
     71             if body.get('type') == 'error.list':
---> 72                 cls.raise_application_errors_on_failure(body, resp.status_code)
     73             return body
     74         except ValueError:

/Users/lihanli/.virtualenvs/mathspace/lib/python2.7/site-packages/intercom/request.pyc in raise_application_errors_on_failure(cls, error_list_details, http_code)
    129         else:
    130             message = error_details['message']
--> 131         raise error_class(message, error_context)
    132 
    133     @classmethod

BadRequestError: Email can't be blank

```

**Comment by asavoy** [comment]
We managed to find a workaround:

```python
user = intercom.User.create(user_id=99999, email=None) # OK

user.increment('logins')
user.save() # BadRequestError -- probably because now user.email == ""

user.email = None
user.save() # OK
```

**Comment by jkeyes** [comment]
The fix will be in the 2.0.1 release.
